pkg/pipeline/transform/transform_generic.go

```diff
@@ -33,17 +33,50 @@ type Generic struct {
 // Transform transforms a flow to a new set of keys
 func (g *Generic) Transform(entry config.GenericMap) (config.GenericMap, bool) {
 	var outputEntry config.GenericMap
+	ok := true
 	log.Tracef("Transform input = %v", entry)
 	if g.policy != "replace_keys" {
 		outputEntry = entry.Copy()
 	} else {
 		outputEntry = config.GenericMap{}
 	}
 	for _, transformRule := range g.rules {
-		outputEntry[transformRule.Output] = entry[transformRule.Input]
+		if transformRule.Multiplier != 0 {
+			switch entry[transformRule.Input].(type) {
+			case int:
+				outputEntry[transformRule.Output] = transformRule.Multiplier * entry[transformRule.Input].(int)
+			case uint:
+				outputEntry[transformRule.Output] = uint(transformRule.Multiplier) * outputEntry[transformRule.Input].(uint)
+			case int8:
+				outputEntry[transformRule.Output] = int8(transformRule.Multiplier) * outputEntry[transformRule.Input].(int8)
+			case uint8:
+				outputEntry[transformRule.Output] = uint8(transformRule.Multiplier) * outputEntry[transformRule.Input].(uint8)
+			case int16:
+				outputEntry[transformRule.Output] = int16(transformRule.Multiplier) * outputEntry[transformRule.Input].(int16)
+			case uint16:
+				outputEntry[transformRule.Output] = uint16(transformRule.Multiplier) * outputEntry[transformRule.Input].(uint16)
+			case int32:
+				outputEntry[transformRule.Output] = int32(transformRule.Multiplier) * outputEntry[transformRule.Input].(int32)
+			case uint32:
+				outputEntry[transformRule.Output] = uint32(transformRule.Multiplier) * outputEntry[transformRule.Input].(uint32)
+			case int64:
+				outputEntry[transformRule.Output] = int64(transformRule.Multiplier) * outputEntry[transformRule.Input].(int64)
+			case uint64:
+				outputEntry[transformRule.Output] = uint64(transformRule.Multiplier) * outputEntry[transformRule.Input].(uint64)
+			case float32:
+				outputEntry[transformRule.Output] = float32(transformRule.Multiplier) * entry[transformRule.Input].(float32)
+			case float64:
+				outputEntry[transformRule.Output] = float64(transformRule.Multiplier) * entry[transformRule.Input].(float64)
+			default:
+				ok = false
+				log.Errorf("%s not of numerical type; cannot perform multiplication", transformRule.Output)
```

Comment:
Should we add an operational metric to track the number of errors of this type?

Reply:
The other transforms do not currently have operational metrics. We should probably make a project-wide decision on whether to output such metrics from the transform stage.